### PR TITLE
TRD: separate pileup implementation from digitizer

### PR DESCRIPTION
**Pull request by jolopezl**
Separate the digitizer's pileup implementation to add a cleaner way to test it; @martenole corrected a bug in the implementation recently. We noted that it wasn't straightforward to add a test. By having a separate tool for the pileup implementation, one doesn't have to instantiate the digitizer for testing (ugly) without making the pileup method public. This was a good sign that this method needed a separate place.

The definition of `SignalArray` needed to be extracted from the Digitizer header to avoid a circular reference (between `PileupTool` and `Digitizer`). I chose to put it under DataFormatsTRD, which seems to be ok.

In summary, the end goal of the change: add a test for the pileup implementation.

### CODE REVIEW

File: Detectors/TRD/simulation/test/testPileupTool.cxx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  double triggerTime;
+  std::vector<SignalArray> signalArrays(3);
+  std::array<SignalContainer, constants::MAXCHAMBER> chamberSignals;
+  std::deque<std::array<SignalContainer, constants::MAXCHAMBER>> pileupSignals;
+
+  // set the signals
+  signalArrays[0].firstTBtime = 0;
+  signalArrays[1].firstTBtime = 2000;
+  signalArrays[2].firstTBtime = 4900;
+  fill_ones(signalArrays[0].signals);
```
Comment (martenole) [inline Detectors/TRD/simulation/test/testPileupTool.cxx]: Fyi: You could just use `std::fill(signalArrays[0].signals.begin(), signalArrays[0].signals.end(), 1)`

**Review by martenole — DISMISSED** [review]
Thanks a lot I think it is good that we have the possibility to test the pileup implementation now.
The only drawback I see at the moment is that the expected output is hardcoded, but it depends on the readout time. The readout time is a constant, but I am not 100% it will never change. And if that happens then the test will fail.
Might be relevant for Kr runs? Or when we recompile with a different number of time bins?

File: Detectors/TRD/simulation/test/testPileupTool.cxx (hunk 1 above)
Comment (jolopezl) [reply Detectors/TRD/simulation/test/testPileupTool.cxx]: Thanks! I forgot about `std::fill`. I will use it instead of that overthought lambda.

**Comment by jolopezl** [comment]
> Thanks a lot I think it is good that we have the possibility to test the pileup implementation now.
> The only drawback I see at the moment is that the expected output is hardcoded, but it depends on the readout time. The readout time is a constant, but I am not 100% it will never change. And if that happens then the test will fail.
> Might be relevant for Kr runs? Or when we recompile with a different number of time bins?

I think both the readout time and the number of time bins should be taken from configuration or the CCDB at some point. The test can be adapted accordingly when those changes come in.

**Review by martenole — APPROVED** [review]
